core/domain/user_services.py

```diff
@@ -111,7 +119,27 @@ def require_valid_username(cls, username):
             # Admin usernames are reserved for admins. Note that 'admin'
             # itself is already in use for the demo exploration.
             raise utils.ValidationError('This username is not available.')
-
+    @classmethod
+    def fetch_gravatar(cls, email):
+        """Gets autogenerated avatar for a given email.
+
+        Returns Default picture if there is an error.
+        """
+        base_url = "http://www.gravatar.com/avatar/"
+        avatar_url = base_url + hashlib.md5(email.lower()).hexdigest() + "?"
+        avatar_url += urllib.urlencode({'d':'retro', 's':str(AVATAR_SIZE)})
```

Comment:
Thats one of the different style gravatars available.
<img width="703" alt="capture" src="https://cloud.githubusercontent.com/assets/10217535/13533805/23c6ed4c-e258-11e5-8e66-7041020a8f89.PNG">